pallets/manta-pay/Cargo.toml

```diff
@@ -87,35 +87,35 @@ precompute-coins = [
 # utils
 anyhow = { version = "1.0.55", optional = true, default-features = false }
 base64 = { version = "0.20", default-features = false, features = ["alloc"] }
-indoc = { version = "1.0.3", optional = true, default-features = false }
-rand_chacha = { version = "0.3.1", optional = true, default-features = false }
-tempfile = { version = "3.3.0", optional = true, default-features = false }
 
 # substrate dependencies
 frame-benchmarking = { git = "https://github.com/paritytech/substrate.git", branch = "polkadot-v0.9.26", default-features = false, optional = true }
 frame-support = { git = "https://github.com/paritytech/substrate.git", branch = "polkadot-v0.9.26", default-features = false }
 frame-system = { git = "https://github.com/paritytech/substrate.git", branch = "polkadot-v0.9.26", default-features = false }
-scale-codec = { package = "parity-scale-codec", version = '3.1.2', default-features = false, features = ["derive", "max-encoded-len"] }
-scale-info = { version = "2.1.2", default-features = false, features = ["derive"] }
-sp-api = { git = 'https://github.com/paritytech/substrate.git', branch = "polkadot-v0.9.26", optional = true, default-features = false }
-sp-blockchain = { git = 'https://github.com/paritytech/substrate.git', branch = "polkadot-v0.9.26", optional = true, default-features = false }
-sp-runtime = { git = "https://github.com/paritytech/substrate.git", branch = "polkadot-v0.9.26", default-features = false }
-sp-std = { git = 'https://github.com/paritytech/substrate.git', branch = "polkadot-v0.9.26", default-features = false }
+indoc = { version = "1.0.3", optional = true, default-features = false }
 
 # JSON-RPC
 jsonrpsee = { version = "0.14.0", features = ["server", "macros"], optional = true }
 
 # manta dependencies
-manta-accounting = { git = "https://github.com/manta-network/manta-rs.git", rev = "249f8d22b9b634602ee5f2c1e6fbad878f33a729", default-features = false }
-manta-crypto = { git = "https://github.com/manta-network/manta-rs.git", rev = "249f8d22b9b634602ee5f2c1e6fbad878f33a729", default-features = false }
-manta-pay = { git = "https://github.com/manta-network/manta-rs.git", rev = "249f8d22b9b634602ee5f2c1e6fbad878f33a729", default-features = false, features = ["groth16", "parameters", "scale"] }
+manta-accounting = { git = "https://github.com/manta-network/manta-rs.git", branch = "fix/distinguish-errors", default-features = false }
```

Comment:
Maybe let's just wait merging this until the changes are in a manta-rs releasse.